### ISSUE TITLE
[BUG] install script does not run in workspace in 8.5.4

### ISSUE THREAD

**Issue by perrin4869** [npm/cli#4552]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

Given the following:

```
❯ cat package.json
{
  "name": "npm854",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "install": "echo root install",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "workspaces": ["subpackage"]
}
❯ cat subpackage/package.json
{
  "name": "npm854-subpackage",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "install": "touch install",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}
```

Running `npm install` will not generate an `install` file inside the `subpackage` directory.

### Expected Behavior

In `8.5.3` it would run the `install` script inside the workspaces as well
Probably introduced by https://github.com/npm/cli/pull/4529

### Environment

- npm: 8.5.4
- Node.js: 16.14.0
- OS Name: slackware64-current
- npm config:
```ini
; "user" config from /home/perrin4869/.npmrc

//npm.pkg.github.com/:_authToken = (protected)
//registry.npmjs.org/:_authToken = (protected)

; node bin location = /usr/bin/node
; cwd = /home/perrin4869/nodejs/npm854
; HOME = /home/perrin4869
; Run `npm config ls -l` to show all defaults.
```

**Comment by n8agrin** [comment]
+1 I have a repro here: https://github.com/n8agrin/npm-8.5.4-no-lifecycle-scripts

It seems all lifecycle scripts in workspaces do not run in npm versions after 8.5.3

**Comment by threema-danilo** [comment]
This also seems to happen if you don't use a workspace, but instead use a `file:`-dependency in `package.json`:

    "eslint-plugin-xyz": "file:libs/eslint-plugin-xyz",

When running `npm install` in the main directory, with npm 8.5.0, the `install` script of the linked plugin is run, with npm 8.5.5 it is not run.

**Comment by ivancuric** [comment]
Issue is still present in npm 8.9.0.

**Comment by ruyadorno** [comment]
should be fixed in `npm@8.10.0`
